float64[] th_up
float64[] th_down